app/src/main/java/org/microg/gms/droidguard/DroidGuardDatabase.java

```diff
@@ -19,12 +19,18 @@ public DroidGuardDatabase(Context context) {
     @Override
     public void onCreate(SQLiteDatabase db) {
         // Note: "NON NULL" is actually not a valid sqlite constraint, but this is what we see in the original database *shrug*
-        db.execSQL("CREATE TABLE main (a TEXT NOT NULL, b LONG NOT NULL, c LONG NOT NULL, d TEXT NON NULL, e TEXT NON NULL,f BLOB NOT NULL,g BLOB NOT NULL);");
+        db.execSQL("CREATE TABLE IF NOT EXISTS main (a TEXT NOT NULL, b LONG NOT NULL, c LONG NOT NULL, d TEXT NON NULL, e TEXT NON NULL,f BLOB NOT NULL,g BLOB NOT NULL);");
```

Comment:
The `onCreate()` of SQLiteOpenHelper is only called after the initial creation of the database, thus making it a `IF NOT EXISTS` should not have any effect as the table can't be present (assuming we come from a clean state). That's the purpose of using SQLiteOpenHelper instead of directly accessing SQLiteDatabase.

Reply:
Sadly DG crashed if "IF NOT EXISTS" isn't present. The file and table existed. On the other hand it's good to have more precautions than less :)

Reply:
Maybe the file existed because you upgraded from your previous version that wasn't using SQLiteOpenHelper? Because it doesn't crash for me.